servers/audio/effects/eq.cpp

```diff
@@ -103,7 +103,12 @@ void EQ::recalculate_band_coefficients() {
 
 		//printf("band %i, precoefs = %f,%f,%f\n",i,c2a,c2b,c2c);
 
-		double r1, r2; //roots
+		// We're assigning r1 an arbitrary value to to avoid the compiler "'r1' may be used uninitialized" warning below.
+		// We're certain it's either initialised in solve_quadratic or solve_quadratic returns 0.
+		// We can't check that the aribtrary value has been changed by solve_quadratic afterwards, because
+		// any arbitrary value can also be a valid root.
+		// We're using an arbitrary value, because we can't currently use std::numeric_limits<double>::quiet_NaN().
+		double r1 = 0, r2; //roots
```

Comment:
What about `r2`? If the compiler only complains about `r1`, this is starting to sound like a compiler bug more than something we need to heavily document.

Reply:
r2 is not used, so it does raise a warning.

Reply:
Right. I would still initialize it in any case.
The general rule is that everything should be initialized, unless it's really in performance critical code and there's a measured impact of default initialization on performance.

Reply:
Done.